setup.py

```diff
@@ -27,14 +27,68 @@
     "rsa>=3.1.4,<5",
 )
 
+# TODO(https://github.com/googleapis/google-auth-library-python/issues/1440): Unit test fails with
+#  `No module named 'cryptography.hazmat.backends.openssl.x509' for Python 3.7``.
+cryptography_base_require = [
+    "cryptography>=38.0.3",
+    "cryptography < 39.0.0; python_version < '3.8'",
+]
+
+requests_extra_require = ["requests >= 2.20.0, < 3.0.0.dev0"]
+
+aiohttp_extra_require = ["aiohttp >= 3.6.2, < 4.0.0.dev0", *requests_extra_require]
+
+pyjwt_extra_require = ["pyjwt>=2.0", *cryptography_base_require]
+
+reauth_extra_require = ["pyu2f>=0.1.5"]
+
+enterprise_cert_extra_require = ["cryptography", "pyopenssl"]
+
+pyopenssl_extra_require = ["pyopenssl>=20.0.0", cryptography_base_require]
+
+urllib3_extra_require = [["urllib3", "packaging"]]
```

Comment:
The minimum version of dependencies should be stated. Once that's done, we can update the constraints file to test the minimum version.
https://github.com/googleapis/google-auth-library-python/blob/main/testing/constraints-3.7.txt